Classes/ScrnM32GrenadeLauncher.uc

```diff
@@ -1,8 +1,372 @@
+    if (ClientState == WS_BringUp || ClientState == WS_ReadyToFire)
+    {
+        if ( (Instigator.PendingWeapon != None) && !Instigator.PendingWeapon.bForceSwitch )
+        {
+            for (Mode = 0; Mode < NUM_FIRE_MODES; Mode++)
+            {
+                // if _RO_
+                if( FireMode[Mode] == none )
+                continue;
+                // End _RO_
+
+                if ( FireMode[Mode].bFireOnRelease && FireMode[Mode].bIsFiring )
+                return false;
+                if ( FireMode[Mode].NextFireTime > Level.TimeSeconds + FireMode[Mode].FireRate*(1.f - MinReloadPct))
+                DownDelay = FMax(DownDelay, FireMode[Mode].NextFireTime - Level.TimeSeconds - FireMode[Mode].FireRate*(1.f - MinReloadPct));
+            }
+        }
+
+        if (Instigator.IsLocallyControlled())
+        {
+            for (Mode = 0; Mode < NUM_FIRE_MODES; Mode++)
+            {
+                // if _RO_
+                if( FireMode[Mode] == none )
+                continue;
+                // End _RO_
+
+                if ( FireMode[Mode].bIsFiring )
+                ClientStopFire(Mode);
+            }
+
+            if (  DownDelay <= 0  || KFPawn(Instigator).bIsQuickHealing > 0)
+            {
+                if ( ClientState == WS_BringUp || KFPawn(Instigator).bIsQuickHealing > 0 )
+                TweenAnim(SelectAnim,PutDownTime);
+                else if ( HasAnim(PutDownAnim) )
+                {
+                    if( ClientGrenadeState == GN_TempDown || KFPawn(Instigator).bIsQuickHealing > 0)
+                    {
+                        PlayAnim(PutDownAnim, PutDownAnimRate * (PutDownTime/QuickPutDownTime), 0.0);
+                    }
+                    else
+                    {
+                        PlayAnim(PutDownAnim, PutDownAnimRate, 0.0);
+                    }
+
+                }
+            }
+        }
+        ClientState = WS_PutDown;
+        if ( Level.GRI.bFastWeaponSwitching )
+        DownDelay = 0;
+        if ( DownDelay > 0 )
+        {
+            SetTimer(DownDelay, false);
+        }
+        else
+        {
+            if( ClientGrenadeState == GN_TempDown )
+            {
+                SetTimer(QuickPutDownTime, false);
+            }
+            else
+            {
+                SetTimer(PutDownTime, false);
+            }
+        }
+    }
+    for (Mode = 0; Mode < NUM_FIRE_MODES; Mode++)
+    {
+        // if _RO_
+        if( FireMode[Mode] == none )
+        continue;
+        // End _RO_
+
+        FireMode[Mode].bServerDelayStartFire = false;
+        FireMode[Mode].bServerDelayStopFire = false;
+    }
+    Instigator.AmbientSound = None;
+    OldWeapon = None;
+    return true; // return false if preventing weapon switch
+}
 
 defaultproperties
 {
      FireModeClass(0)=Class'ScrnBalanceSrv.ScrnM32Fire'
+     ReloadRate=10.204 //added to support new reload
+     BulletLoadRate=1.50 // increased base reload speed from original value 1.634
```

Comment:
I'm not insisting on that, but how about setting BulletLoadRate=1.40? Then it will have the same load time on 2 rounds (now it is 3) and winning 1s when roading all 6 rounds.

Reply:
Actually I had thought of setting it for same load time on 2 rounds as well, I will test with the new BulletLoadRate, speed up the animation to match and also write a new value for ReloadRate(I think it does something) and if it's still synced to the reload nicely then I'll change them.

Reply:
I like what you are doing Scrndles :) :+1: